app/screens/apps_form/apps_form.tsx

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import {ScrollView} from 'react-native';
+import {EventSubscription, Navigation} from 'react-native-navigation';
+import {SafeAreaView} from 'react-native-safe-area-context';
+
+import {checkDialogElementForError, checkIfErrorsMatchElements} from '@mm-redux/utils/integration_utils';
+
+import ErrorText from 'app/components/error_text';
+import StatusBar from 'app/components/status_bar';
+import FormattedText from 'app/components/formatted_text';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {dismissModal} from 'app/actions/navigation';
+
+import DialogIntroductionText from './dialog_introduction_text';
+import {Theme} from '@mm-redux/types/preferences';
+import {AppCall, AppCallResponse, AppField, AppForm, AppFormValue, AppFormValues, AppSelectOption} from '@mm-redux/types/apps';
+import {DialogElement} from '@mm-redux/types/integrations';
+import {AppCallResponseTypes} from '@mm-redux/constants/apps';
+import AppsFormField from './apps_form_field';
+
+export type Props = {
+    call: AppCall;
+    form: AppForm;
+    actions: {
+        submit: (submission: {
+            values: {
+                [name: string]: string;
+            };
+        }) => Promise<{data: AppCallResponse<FormResponseData>}>;
+        performLookupCall: (field: AppField, values: AppFormValues, userInput: string) => Promise<AppSelectOption[]>;
+        refreshOnSelect: (field: AppField, values: AppFormValues, value: AppFormValue) => Promise<{data: AppCallResponse<any>}>;
+    };
+    theme: Theme;
+    componentId: string;
+}
+
+type FormResponseData = {
+    errors: {
+        [field: string]: string;
+    };
+}
+
+type State = {
+    values: {[name: string]: string};
+    error: string | null;
+    errors: {[name: string]: React.ReactNode};
+    submitting: boolean;
+    form: AppForm;
+}
+
+const initFormValues = (form: AppForm): {[name: string]: string} => {
+    const values: {[name: string]: any} = {};
+    if (form && form.fields) {
+        form.fields.forEach((f) => {
+            values[f.name] = f.value || null;
+        });
+    }
+
+    return values;
+};
+
+export default class AppsForm extends PureComponent<Props, State> {
+    private scrollView: React.RefObject<ScrollView>;
+    private navigationEventListener?: EventSubscription;
+    private submitted = false;
+
+    constructor(props: Props) {
+        super(props);
+
+        const {form} = props;
+        const values = initFormValues(form);
+
+        this.state = {
+            values,
+            error: null,
```

Comment:
could we use one `errors` set to capture all errors (and remove the unique `error` ?)